### ISSUE TITLE
css-tree reporting "SyntaxError: ")" is expected" 

### ISSUE THREAD

**Issue by himynameisdave** [stylelint/stylelint#6798]
### What steps are needed to reproduce the bug?

- Install `stylelint` + `less` deps:
  ```
  yarn add stylelint@^15.5.0 stylelint-config-standard@^33.0.0 less@^4.1.3 postcss-less@^6.0.0
  ```
- Create a `.stylelintrc.json` file which looks like the one below.
- Run `stylelint "**/*.less|**/*.css"`

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "extends": "stylelint-config-standard",
  "overrides": [
    {
      "files": ["*.less"],
      "customSyntax": "postcss-less"
    }
  ],
  "ignoreFiles": ["dist/**/*"],
  "rules": {
    "selector-class-pattern": null,
    "no-descending-specificity": null,
    "function-no-unknown": [true, {
      "ignoreFunctions": [
        "fade",
        "lighten",
        "darken"
      ]
    }]
  }
}
```

### How did you run Stylelint?

stylelint "**/*.less|**/*.css"

### Which version of Stylelint are you using?

^15.5.0

### What did you expect to happen?

Stylelint to run, as it did with `^15.4.0`.

### What actually happened?

Cryptic error stemming from `css-tree`:

```
SyntaxError: ")" is expected
    at TokenStream.error (/Users/himynameisdave/stylelint-project/node_modules/css-tree/cjs/parser/create.cjs:277:19)
    at TokenStream.eat (/Users/himynameisdave/stylelint-project/node_modules/css-tree/cjs/parser/create.cjs:218:22)
    at TokenStream.parse [as Function] (/Users/himynameisdave/stylelint-project/node_modules/css-tree/cjs/syntax/node/Function.cjs:24:14)
    at TokenStream.defaultRecognizer (/Users/himynameisdave/stylelint-project/node_modules/css-tree/cjs/syntax/scope/default.cjs:41:24)
    at TokenStream.readSequence (/Users/himynameisdave/stylelint-project/node_modules/css-tree/cjs/parser/sequence.cjs:24:40)
    at TokenStream.parse [as Function] (/Users/himynameisdave/stylelint-project/node_modules/css-tree/cjs/syntax/node/Function.cjs:21:24)
    at TokenStream.defaultRecognizer (/Users/himynameisdave/stylelint-project/node_modules/css-tree/cjs/syntax/scope/default.cjs:41:24)
    at TokenStream.readSequence (/Users/himynameisdave/stylelint-project/node_modules/css-tree/cjs/parser/sequence.cjs:24:40)
```

I was able to get additional data by opening `node_modules/css-tree/cjs/parser/create.cjs` and adding additional console.logs. By doing so I was able to see where the error was coming from, I _think_ it was choking on two things, both of which are some basic Less nesting syntax:

```less
.linear-gradient(@start, @end) {
  background: -webkit-gradient(linear, center top, center bottom, from(@start), to(@end));
  background: -webkit-linear-gradient(@start, @end);
  background: -moz-linear-gradient(@start, @end);
  background: -o-linear-gradient(@start, @end);
  background: -ms-linear-gradient(@start, @end);
  background: linear-gradient(@start, @end);
}
```

```less
.margin {
  &-right {
    &-sm {
      margin-right: 5px;
    }
    // ...
  }
}

```

### Does the bug relate to non-standard syntax?

Yes, it's Less

### Proposal to fix the bug

I have no idea. I tried digging into this to figure out what might be broken. Apologies if this is filed in the wrong place, I suspect this might be a `postcss-less` issue but I'm not sure.

**Comment by yoyo837** [comment]
See #6791

**Comment by ybiquitous** [comment]
@himynameisdave Thanks for the report. As you @yoyo837 commented, this is a duplicate of #6791. And we will release the fix in a few days.
